cmd/deploy.go

```diff
@@ -111,27 +111,32 @@ func runDeploy(cmd *cobra.Command, _ []string, clientFn deployClientFn) (err err
 		return err
 	}
 
-	config, err = config.Prompt()
-	if err != nil {
-		if err == terminal.InterruptErr {
-			return nil
-		}
-		return
-	}
-
 	function, err := functionWithOverrides(config.Path, functionOverrides{Namespace: config.Namespace, Image: config.Image})
 	if err != nil {
 		return
 	}
 
-	function.Envs, err = mergeEnvs(function.Envs, config.EnvToUpdate, config.EnvToRemove)
-	if err != nil {
-		return
+	if len(function.Envs) > 0 {
+		function.Envs, err = mergeEnvs(function.Envs, config.EnvToUpdate, config.EnvToRemove)
+		if err != nil {
+			return
+		}
 	}
 
-	// Check if the Function has been initialized
-	if !function.Initialized() {
-		return fmt.Errorf("the given path '%v' does not contain an initialized function. Please create one at this path before deploying", config.Path)
+	if config.Build {
```

Comment:
This is a good idea, but I think it should actually be part of the `Prompt` method.  For two reasons, first, this causes the user to never be prompted for other attributes, such as Namespace, which are pertinent whether or not build is enabled.  Second, the purpose of this `if` statement is to adjust how `Prompt` is implemented, so it should be part of Prompt's implementation.

The way this is can be handled is by splitting Prompt into multiple sets of queries (you can see this in action in `invoke.go` which only asks for `Data` if `File` was not specified.